### ISSUE TITLE
No timeout error if authority is not reachable

### ISSUE THREAD

**Issue by franzk** [AzureAD/microsoft-authentication-library-for-js#1306]
## Library
- [x] `msal@1.2.1`

## Description
There's no timeout error if the authority is not reachable. `WindowUtils::monitorWindowForHash` loops forever, because https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/c7c5d1e2c88f7bcbf6f108099f8fc06350773223/lib/msal-core/src/utils/WindowUtils.ts#L61 always matches. I'd consider a timeout to be a hard limit, whatever happens. I came to this issue because Safari 13's ITP blocks the iframe's request and the same thing happens, without any timeout.

## Security
No.

## Regression
Dunno.

## Configuration

```js
{
  auth: {
    authority: 'https://login.microsoftonline.com/' +
      process.env.REACT_APP_TENANT + '/',
    clientId: process.env.REACT_APP_CLIENT_ID,
    navigateToLoginRequestUrl: false,
    redirectUri: window.location.origin
  },
  cache: {
    cacheLocation: 'localStorage'
  },
  system: {
    loadFrameTimeout: 15000
  }
};
```

## Reproduction steps
1. Add this entry to your /etc/hosts file:
```
127.0.0.1 login.microsoftonline.com
```
2. Call `acquireTokenSilent`

## Expected behavior
I expect a timeout error after the configured period (default 6000 ms).

## Browsers
Happens in my Chromium 80.0.3987.122 (Official Build) Arch Linux (64-bit). Firefox (73.0.1) at least throws "ClientAuthError: Error: could not resolve endpoints. Please check network and try again. Details: 0" earlier in the process.

**Comment by franzk** [comment]
BTW: My workaround is to have my own timeout around `acquireTokenSilent`, but this is silly, as the loop keeps running and the iframe remains.

**Comment by AndreiZe** [comment]
We ran into this as well. 

We are seeing this issue when silent token fetch is called and there is network issue. In addition, there has to be cached response for openid-configuration endpoint; otherwise, endpoints_resolution_error is returned. I believe that the issue is due to the fact that with network error hidden iframe that is used to get token response from login.microsoftonline.com never gets redirected back to reply url. The code that Franz references is the exact spot where issue happens: getting contentWindow.location.href will always fail due to CORS. This results into internal MSAL timeout logic being bypassed. 

This is very serious issue. 

First, this creates infinite loop so token fetch never completes.  
Second, this results in triggering logic which checks for iframe every 50 ms. If your app needs many different tokens then this results in multiple timers being triggered every 50ms. 
Third, I am not sure if implementing app specific timeout sufficiently mitigates the issue. MSAL creates whole bunch temporary variables to track ongoing token fetches. When there is request for the same scopes that are already being tracked then MSAL simply waits for original request to complete which never does. I think it is possible to work around by specifying unique state inside token request but this results in slow memory leak. Moreover MSAL creates temporary local storage entries for each token fetch and these will not get cleared, eventually resulting in local storage being exhausted.

**Comment by jasonnutter** [comment]
The fix for this is available in `msal@1.2.2-beta.2`, please try that version and confirm that it has been fixed, thanks!
